crates/flux-refineck/src/primops.rs

```diff
@@ -316,6 +389,20 @@ fn mk_sub_rules(overflow_mode: OverflowMode) -> RuleMatcher<2> {
             }
         }
 
+        // like Lazy, but we also check for underflow on unsigned subtraction
+        OverflowMode::StrictUnder => {
+            primop_rules! {
+                fn(a: T, b: T) -> T{v: E::implies(valid_int(a - b, int_ty), E::eq(v, a - b)) }
+                if let &BaseTy::Int(int_ty) = T
+
+                fn(a: T, b: T) -> T{v: E::implies(valid_uint(a - b, uint_ty), E::eq(v, a - b)) }
+                requires nonnegative(a - b) => ConstrReason::Overflow // check for underflow
```

Comment:
Since we are at this, maybe we can have a dedicated reason, i.e., an error message. The one you get right talks about overflow, which can be confusing.